pkg/alertmanager/multitenant_test.go

```diff
@@ -601,6 +608,97 @@ func TestMultitenantAlertmanager_deleteUnusedLocalUserState(t *testing.T) {
 	require.NotZero(t, dirs[user2]) // has config, files survived
 }
 
+func TestMultitenantAlertmanager_zoneAwareSharding(t *testing.T) {
```

Comment:
Really nice job adding the test :) Made it really clear to me the effect of the change.